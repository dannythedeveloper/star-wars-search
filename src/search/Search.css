.search-form {
    background-color: gray;
    padding: 10px;
    text-align: center;
}

.results-filter {
    background-color: lightgray;
    padding: 10px;
    text-align: center;
}

.book-type {
    margin-left: 5px;
}