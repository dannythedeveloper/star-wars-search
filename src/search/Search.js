import React from 'react';
import './Search.css';

export default function Search(props) {
    return (
        <div className='search'>
            <form 
                className='search-form' 
                onSubmit={props.getSearchInfo}
            >
                <label htmlFor='search-bar' className='search'>Search: </label>
                <input
                    id='search-bar'
                    name='search-bar'
                    className='search-bar'
                    type='text'
                    onChange={props.handleSearch}
                    aria-required='true'
                />
                <label htmlFor='filter-select' className='filter'>Filter: </label>
                <select 
                    id='filter-select'
                    name='filter-select'
                    className='filter-select' 
                    onChange={props.handleFilter} 
                    aria-required='true'
                >
                    <option value=''>Select One</option>
                    <option value='planets'>Planets</option>
                    <option value='starships'>Starships</option>
                    <option value='vehicles'>Vehicles</option>
                    <option value='people'>Characters</option>
                    <option value='films'>Films</option>
                    <option value='species'>Species</option>
                </select>
                <button type="submit">Search</button>
            </form>
        </div>
    )
}