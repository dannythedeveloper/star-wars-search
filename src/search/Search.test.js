import React from 'react';
import ReactDOM from 'react-dom';
import Search from './Search';
import renderer from 'react-test-renderer';

describe('Search Component', () => {
    it('render without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Search />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI', () => {
        const tree = renderer.create(<Search />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});