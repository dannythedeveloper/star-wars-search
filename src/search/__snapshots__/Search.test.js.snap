// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component renders the UI 1`] = `
<div
  className="search"
>
  <form
    className="search-form"
  >
    <label
      className="search"
      htmlFor="search-bar"
    >
      Search: 
    </label>
    <input
      className="search-bar"
      required={true}
      type="text"
    />
    <label
      className="filter"
      htmlFor="filter-select"
    >
      Filter: 
    </label>
    <select
      className="filter-select"
      required={true}
    >
      <option
        value=""
      >
        Select One
      </option>
      <option
        value="planets"
      >
        Planets
      </option>
      <option
        value="starships"
      >
        Starships
      </option>
      <option
        value="vehicles"
      >
        Vehicles
      </option>
      <option
        value="people"
      >
        Characters
      </option>
      <option
        value="films"
      >
        Films
      </option>
      <option
        value="species"
      >
        Species
      </option>
    </select>
    <button
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;
